nntrainer/dataset/random_data_producers.cpp

```diff
@@ -169,4 +174,70 @@ RandomDataOneHotProducer::finalize(const std::vector<TensorDim> &input_dims,
     return DataProducer::Iteration(false, inputs, labels);
   };
 }
+
+DataProducer::Generator_sample RandomDataOneHotProducer::finalize_sample(
+  const std::vector<TensorDim> &input_dims,
```

Comment:
I thought it is also similar to layer implementation. we have this, ```finailze_sample()``` with ```finalize()``` is because we have both sample based and batch based implementation? and it will be the same for other producers as well?

Reply:
`finalize()` will be deleted and `finalize_sample()` will be renamed to `finalize()`

The reason being we have two function is to make a gradual change :)